titus-server-gateway/src/main/java/com/netflix/titus/gateway/service/v3/internal/GatewayJobServiceGateway.java

```diff
@@ -313,14 +314,25 @@ public GatewayJobServiceGateway(GrpcRequestConfiguration tunablesConfiguration,
     private Observable<TaskQueryResult> newActiveTaskQueryAction(TaskQuery taskQuery, CallMetadata callMetadata) {
         if (localCacheQueryProcessor.canUseCache(taskQuery.getFilteringCriteriaMap(), "findTasks", callMetadata)) {
             return localCacheQueryProcessor.syncCache("findTasks", TaskQueryResult.class).concatWith(
-                    Observable.fromCallable(() -> localCacheQueryProcessor.findTasks(taskQuery))
+                    Observable.fromCallable(() -> {
+                        TaskQueryResult taskQueryResult = localCacheQueryProcessor.findTasks(taskQuery);
+                        return taskQueryResult.toBuilder()
+                                .clearItems()
+                                .addAllItems(taskDataInjector.injectIntoTasks(taskQueryResult.getItemsList()))
+                                .build();
+                    })
             );
         }
 
-        return createRequestObservable(emitter -> {
+        return GrpcUtil.<TaskQueryResult>createRequestObservable(emitter -> {
             StreamObserver<TaskQueryResult> streamObserver = createSimpleClientResponseObserver(emitter);
             createWrappedStub(client, callMetadata, tunablesConfiguration.getRequestTimeoutMs()).findTasks(taskQuery, streamObserver);
-        }, tunablesConfiguration.getRequestTimeoutMs());
+        }, tunablesConfiguration.getRequestTimeoutMs()).map(taskQueryResult ->
+                taskQueryResult.toBuilder()
```

Comment:
This looks really similar to `injectIntoTaskQueryResult`, can that function be used here? (I don't fully understand what is happening here with .map()

Reply:
Good catch. I can simplify this code.